tasks/Task.cpp

```diff
@@ -301,13 +341,135 @@ base::samples::RigidBodyState Task::convertToRbs(vpHomogeneousMatrix pose)
     rbs.position[0] = pose.getTranslationVector()[0];
     rbs.position[1] = pose.getTranslationVector()[1];
     rbs.position[2] = pose.getTranslationVector()[2];
-    
+
     vpRotationMatrix R;
     pose.extract(R);
-    
     vpQuaternionVector q(R);
-    
     rbs.orientation = base::Orientation(q.w(), q.x(), q.y(), q.z());
 
     return rbs;
 }
+
+void Task::transformInput(base::LinearAngular6DCommand cmd_in_body)
+{
+    //initialize setpoint on the body frame
+    //since the non-expected input are NaN, set it to default values
+    //in order to have valid values for the HomogeneusMatrix creation
+    for (int i = 0; i < 3; ++i)
+    {
+        if (!expected_inputs.linear[i]) {cmd_in_body.linear[i] = 0;}
+        if (!expected_inputs.angular[i])
+        {
+            if ( i == 0 || i == 1)
+            {
+                cmd_in_body.angular[i] = -M_PI_2;
+            }
+            else
+            {
+                cmd_in_body.angular[i] = 0;
+            }
+        }
+    }
+
+    //create a vpHomogeneousMatrix of the object desired position wrt body (bdMo)
+    vpTranslationVector bdto(cmd_in_body.linear[0], cmd_in_body.linear[1], cmd_in_body.linear[2]);
+    vpRxyzVector bdro(cmd_in_body.angular[0], cmd_in_body.angular[1], cmd_in_body.angular[2]);
+    vpRotationMatrix bdRo(bdro);
+    //pose of the object wrt the body
+    vpHomogeneousMatrix bdMo(bdto, bdRo);
+
+    //bdMo is the desired pose of the object in the body frame
+    //cdMo is the desired pose of the object in the camera frame
+    //cMb is the transformation matrix of the body in the camera frame
+    cdMo = cMb * bdMo;
```

Comment:
> WAITING_VALID_TARGET

You can wait for a target. You should complain loudly (i.e. raise an exception *and terminate*) if you get an *invalid* target.